openlibrary/catalog/add_book/load_book.py

```diff
@@ -92,26 +152,39 @@ def walk_redirects(obj, seen):
             seen.add(obj['key'])
         return obj
 
-    q = {'type': '/type/author', 'name': name}  # FIXME should have no limit
-    reply = list(web.ctx.site.things(q))
+    # Try for an 'exact' (case-insensitive) name match, but fall back to alternate_names,
+    # then last name with identical birth and death dates (that are not themselves `None`).
+    queries = [
+        {"type": "/type/author", "name~": author["name"]},
```

Comment:
I should have asked more about this at the time, but which asterisks? If the asterisk is part of an author's name, it will stay as a literal asterisk, and that's the only part of the query that I think is taken as input. E.g., if I have an author named `Catalina Martínez Muñ*z` with the Author ID of `/authors/OL13A`, and I have the code is it is in this commit, then a query using the temporary test endpoint matches her:
```
❯ curl -X POST http://localhost:8080/api/author \
    -H "Content-Type: application/json" \
    -b ~/cookies.txt \
    -d '{
    "authors": [{"name": "Catalina Martínez Muñ*z"}]
}'
{'type': {'key': '/type/edition'}, 'authors': [<Author: '/authors/OL13A'>]}

❯ curl -X POST http://localhost:8080/api/author \
    -H "Content-Type: application/json" \
    -b ~/cookies.txt \
    -d '{
    "authors": [{"name": "Muñ*z", "birth_date": "1900", "death_date": "2000"}]
}'
{'type': {'key': '/type/edition'}, 'authors': [<Author: '/authors/OL13A'>]}

# No match when replacing the middle name with an asterisk
❯ curl -X POST http://localhost:8080/api/author \
    -H "Content-Type: application/json" \
    -b ~/cookies.txt \
    -d '{
    "authors": [{"name": "Catalina*Muñ*z"}]
}'
{'type': {'key': '/type/edition'}, 'authors': [{'type': {'key': '/type/author'}, 'name': 'Catalina*Muñ*z'}]}
```